velox/connectors/hive/HivePartitionFunction.cpp

```diff
@@ -76,7 +76,9 @@ inline uint32_t hashOne<TypeKind::INTEGER>(const int32_t& value) {
 
 template <>
 inline uint32_t hashOne<TypeKind::REAL>(const float& value) {
-  return static_cast<uint32_t>(*reinterpret_cast<const int32_t*>(&value));
+  uint32_t ret;
+  memcpy(&ret, &value, sizeof ret);
```

Comment:
Can you just fix the signature of `hashOne` to pass by value?

Reply:
std::memcpy

Reply:
I believe just fixing the signature is enough without the `std::memcpy`. This should not have any strict aliasing issues.